### PR TITLE
Add SECURITY.md

### LINKED ISSUE #2191: Create a security disclosure policy + process (SECURITY.md)
A CNCF graduation requirement is having a security disclosure process.

Feel free to look at examples in the CNCF ecosystem https://github.com/envoyproxy/envoy/blob/main/SECURITY.md

Other prior art from Rook: https://github.com/rook/rook/blob/master/SECURITY.md

### PR DESCRIPTION
**Pull request by phisco**
### Description of your changes

Adding a `SECURITY.md` file, mainly adapted from containerd's [one](https://github.com/containerd/project/blob/main/SECURITY.md), to enable vulnerability reporting as described [here](https://docs.github.com/en/code-security/getting-started/adding-a-security-policy-to-your-repository).

Fixes #2191 

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

N.A.

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: SECURITY.md
Comment (phisco) [inline SECURITY.md:14]: @jbw976 suggestions on how to get this email address in place?

File: SECURITY.md
Comment (phisco) [inline SECURITY.md:59]: Same as above, might be too much at the moment

File: SECURITY.md
Comment (phisco) [inline SECURITY.md:38]: Not sure whether we want to keep this, we could remove it for the time being and add it later on. WDYT?

**Review by jbw976 — COMMENTED** [review]
Looks like this addresses an old issue we have for adding a security disclosure process, so I've added that to the body: #2191 

This will be helpful to our graduation effort also :)

I'll want to look at this fairly thoroughly, thanks for your patience @phisco!

**Comment by jbw976** [comment]
Also some prior art for how we handle this in the Rook project, which is fully graduated: https://github.com/rook/rook/blob/master/SECURITY.md

File: SECURITY.md
Comment (negz) [inline SECURITY.md:19]: 24 business hours? If we really do mean 24 hours we'll need to attach this email to a pager rotation.

File: SECURITY.md
Comment (negz) [inline SECURITY.md:23]: Perhaps maintainer rather than committer?

File: SECURITY.md
Comment (negz) [reply SECURITY.md:38]: I'd err toward adding later; I think we're only staffed for lightweight/minimal process at the moment.

File: SECURITY.md
Comment (jbw976) [inline SECURITY.md:10]: do we want to link directly to https://github.com/crossplane/crossplane/security here? that's where the security tab for this repo goes to.

File: SECURITY.md
Comment (jbw976) [inline SECURITY.md:3]: I would up front credit where this policy came from (containerd) - see how Rook did this at the very front, crediting Envoy and CoreDNS: https://github.com/rook/rook/blob/master/SECURITY.md#security-release-process

File: SECURITY.md
Comment (jbw976) [reply SECURITY.md:14]: I can follow up to make this email address happen after merging this PR - just remind me :)

File: SECURITY.md
Comment (jbw976) [reply SECURITY.md:19]: I'm OK with this, with the language update to "can typically expect a response within 24 hours acknowledging..."

Since this is a typical expectation (not a promise), and there is a fallback (ping a maintainer directly), I am OK with not having a pager rotation for this.  I've tried to be very responsive for security disclosures on other projects in the past, and giving an initial ACK quickly is something I'm OK with, since it's not a SLA or guarantee.

File: SECURITY.md
Comment (jbw976) [reply SECURITY.md:23]: +1, can we update each usage of 'committer" to be "maintainer" instead? that's the formal position we recognize in our governance.

File: SECURITY.md
Comment (jbw976) [inline SECURITY.md:34]: we'll need to follow up with the CNCF to get this email address created too.

Perhaps we can make the reporting email symmetrical with the announcement email?  e.g. let's just create both on lists.cncf.io?

File: SECURITY.md
Comment (jbw976) [reply SECURITY.md:59]: I'm fine with anyone requesting to join the announcement list - I don't see a need at this point to restrict it heavily since it will be used to broadcast vulnerabilities along with their fixes, to get people to update to the fix.

Maybe we can shorten this to:

```
Any organization or individual wishing to join the security announce mailing list, please send a request to `security@crossplane.io`.
```

**Review by jbw976 — COMMENTED** [review]
This is a great start @phisco, thanks for driving this - let me know if you want to sync on any of the feedback :)

File: SECURITY.md
Comment (jbw976) [inline SECURITY.md:8]: This Github provided process to private report a vulnerability and collaborate on the fix privately is quite nice.  I don't think this was available for us in the past when creating security policies for other projects.  Thanks for finding this, it does look like a nice workflow!!

File: SECURITY.md
Comment (phisco) [reply SECURITY.md:59]: If I got it right, the point here is not to have vulnerabilities disclosed alongside the announcement of the fixed releases, but to have a mailing list from which subscribed users could get notified with vague descriptions of vulnerabilities not yet public, so that they can be aware something is coming. On one side it should not be considered secret as it is not and secrecy is not a valid security strategy, but it should not be too much advertised either.

File: SECURITY.md
Comment (phisco) [reply SECURITY.md:34]: I'm not sure what's the procedure to handle such lists under the cncf, but while the announcement list is a public thing and people should be able to subscribe or be subscribed easily, `security@crossplane.io` should only send emails to a slowly changing set of people, maintainers only most probably, so, it might be easier to handle it having it totally under our control.

Comment (phisco) [reply SECURITY.md:39]: No strong opinions on that though, Rook is also already using "symmetrical" [emails](https://github.com/rook/rook/blob/master/SECURITY.md#mailing-lists).

File: SECURITY.md
Comment (phisco) [reply SECURITY.md:13]: Thanks to @AdamKorcz for pointing that out!

File: SECURITY.md
Comment (jbw976) [reply SECURITY.md]: ah OK @phisco, I think I didn't catch that intention of the announcement list. To make it's intent more clear, perhaps we can add some more language to the beginning of this paragraph like:

> The security announcement list will be used to announce vulnerabilities, often ahead of when a fix is made available, to a restricted set of Crossplane adopters and vendors.

And with that, I'm fine with the restrictions for joining as currently proposed :)

File: SECURITY.md
Comment (jbw976) [reply SECURITY.md:39]: I thought about this a bit more, and I think having all security lists hosted by the CNCF is the right thing to do.  Those lists can be completely restricted as we see fit, and we either have self-service to make those changes ourselves, or updates to the lists are a quick CNCF serviceDesk ticket away.

File: SECURITY.md
Comment (jbw976) [inline SECURITY.md:24]: we can link "maintainer" here to https://github.com/crossplane/crossplane/blob/master/OWNERS.md#maintainers to make it more discoverable :)

File: SECURITY.md
Comment (jbw976) [reply SECURITY.md]: I see now that there is already some language regarding this announcement list in the "review process" section, so you can choose whether you want to add this extra language in the "Joining the security announce mailing list" section or not 🤪

File: SECURITY.md
Comment (jbw976) [inline SECURITY.md:38]: What is the "public disclosure" process or channel? That's what I thought the announcement list had been for, but that's not the case.

Release notes? 🤔

Comment (phisco) [reply SECURITY.md:40]: Added a section for that, the assumption was that it was done via Github security advisories, but it's better to make it explicit.

File: SECURITY.md
Comment (phisco) [reply SECURITY.md]: added, better to be explicit 😉

File: SECURITY.md
Comment (phisco) [reply SECURITY.md]: changed wording as suggested

File: SECURITY.md
Comment (AdamKorcz) [inline SECURITY.md:19]: This may be very rigid, but do you only have to detail the issue and steps to reproduce if you send an email? 

I suggest detailing in a separate section what to include in the report and not as part of where to send the report.

File: SECURITY.md
Comment (AdamKorcz) [inline SECURITY.md:47]: I would recommend adding a line to urge reporters to not discuss issues before Crossplane have announced them.

Comment (AdamKorcz) [reply SECURITY.md:47]: Wait, it is already added in the previous section.

File: SECURITY.md
Comment (phisco) [reply SECURITY.md]: Thanks for the suggestion @AdamKorcz! I tried to add it in a separate section, feel free to give it a look and suggest any other detail that you feel might be worth mentioning!

**Comment by phisco** [comment]
@negz @jbw976 this could be ready for another review 🙏

File: SECURITY.md
Comment (jbw976) [reply SECURITY.md:60]: @phisco I have opened a CNCF service desk ticket to get these mailing lists created!

https://cncfservicedesk.atlassian.net/servicedesk/customer/portal/1/CNCFSD-1597 (this link is probably not accessible for most folks)

File: SECURITY.md
Comment (jbw976) [inline SECURITY.md:22]:
```suggestion
2. Send an email to `crossplane-security@lists.cncf.io` detailing the issue, see below
```

File: SECURITY.md
Comment (jbw976) [inline SECURITY.md:34]:
```suggestion
understand and prioritize it, for example here is a list of details that might be
```

File: SECURITY.md
Comment (jbw976) [inline SECURITY.md:59]:
```suggestion
on public disclosure before the patch release, an announcement will be sent to
```

**Review by jbw976 — APPROVED** [review]
I'll let you know when the new security mailing lists have been created - otherwise, just some small typos to clean up :)

Also consider using capitalized Crossplane everywhere, as mentioned in https://docs.crossplane.io/contribute/writing-style-guide/ 🤓

**Comment by jbw976** [comment]
Good news @phisco! The mailings lists are all set up now by the CNCF. We had to make a few adjustments to the names and the access permissions, but we're good to go now!

Do you want to squash the commit history here before merging? What do you think?

**Comment by phisco** [comment]
Awesome, @jbw976! Squashed everything, we are good to go!

**Comment by jbw976** [comment]
All merged now @phisco, thanks for driving this! Please follow up with me directly if you need anything else in the short term to finish off any fuzz testing related bugs 🙇

**Pull request merged**
